### ISSUE TITLE
differentiable node implementations

### ISSUE THREAD

**Issue by lacava** [lacava/feat#99]
implement differentiable versions of 

- [ ] add
- [ ] subtract
- [ ] multiply
- [ ] divide
- [ ] sin
- [ ] cos
- [ ] exp
- [ ] exponential
- [ ] log
- [ ] square
- [ ] cube
- [ ] sqrt
- [ ] softmax
- [ ] sign
- [ ] logit
- [ ] tanh
- [ ] gaussian
- [ ] 2d gaussian
- [ ] relu - *needs to be implemented from scratch* !
